exp/query/cache.go

```diff
@@ -161,3 +188,35 @@ type TenantCache struct {
 func (tc *TenantCache) AbsolutePath() string {
 	return path.Join(tc.basePath, tc.Collection, tc.TenantID, fmt.Sprintf("%d", tc.Version))
 }
+
+// CacheMetrics exposes some insights about how cache operations.
+type CacheMetrics struct {
+	// OpsDuration tracks overall duration of each cache operation.
+	OpsDuration *prometheus.HistogramVec
+	CacheMiss   *prometheus.CounterVec
```

Comment:
is there a way to infer the cache hit rate from `CacheMiss` so that we can visualize it in grafana? if not, is there a reason not to keep track of both?

Reply:
Yes. it's common to track TotalOps and CacheMiss and we can infer cache-hit from those.

Here total ops comes from `OpsDuration` (it's histogram, so it comes with both latency and count)